compiler/zrc_codegen/src/ty.rs

```diff
@@ -105,7 +105,7 @@ pub fn llvm_int_type<'ctx: 'a, 'a>(
                 &ctx.target_machine().get_target_data(),
                 Some(AddressSpace::default()),
             ),
-            Type::Ptr(_) | Type::Fn(_) | Type::Struct(_) | Type::Union(_) => {
+            Type::Ptr(_) | Type::Fn(_) | Type::Struct(_) | Type::Union(_) | Type::Opaque(_) => {
                 panic!("not an integer type")
```

Comment:
I would much rather this panic with the same error about opaque types being invalid. It is technically not an int, but still, it shouldn't be there at all.